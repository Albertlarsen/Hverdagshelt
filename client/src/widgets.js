// @flow
/* eslint eqeqeq: "off" */

import * as React from 'react';
import { Component } from 'react-simplified';

/**
 * Renders alert messages using Bootstrap classes.
 */
export class Layout extends Component<{windowWidth: number,layoutText?: React.Node, middleWidth: number, children: React.Node}>{
    render(){
        return(
            <ContainerFluid>
                <Row>
                    <Column width={this.props.windowWidth}>
                        <div className="text-center">
                            {this.props.layoutText}
                        </div>
                    </Column>
                    <Column width={this.props.middleWidth}>
                        {this.props.children}
                    </Column>
                    <Column width={this.props.windowWidth}>
                        <div className="text-center">
                            {this.props.layoutText}
                        </div>
                    </Column>
                </Row>
            </ContainerFluid>
        )
    }
}

export class Alert extends Component {
  alerts: { text: React.Node, type: string }[] = [];

  render() {
    return (
      <>
        {this.alerts.map((alert, i) => (
          <div key={i} className={'alert alert-' + alert.type} role="alert">
            {alert.text}
            <button
              className="close"
              onClick={() => {
                this.alerts.splice(i, 1);
              }}
            >
              &times;
            </button>
          </div>
        ))}
      </>
    );
  }

  static success(text: React.Node) {
    // To avoid 'Cannot update during an existing state transition' errors, run after current event through setTimeout
    setTimeout(() => {
      for (let instance of Alert.instances()) instance.alerts.push({ text: text, type: 'success' });
    });
  }

  static info(text: React.Node) {
    // To avoid 'Cannot update during an existing state transition' errors, run after current event through setTimeout
    setTimeout(() => {
      for (let instance of Alert.instances()) instance.alerts.push({ text: text, type: 'info' });
    });
  }

  static warning(text: React.Node) {
    // To avoid 'Cannot update during an existing state transition' errors, run after current event through setTimeout
    setTimeout(() => {
      for (let instance of Alert.instances()) instance.alerts.push({ text: text, type: 'warning' });
    });
  }

  static danger(text: React.Node) {
    // To avoid 'Cannot update during an existing state transition' errors, run after current event through setTimeout
    setTimeout(() => {
      for (let instance of Alert.instances()) instance.alerts.push({ text: text, type: 'danger' });
    });
  }
}
