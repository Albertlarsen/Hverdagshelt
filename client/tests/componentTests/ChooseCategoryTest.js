//@flow

import * as React from 'react';
import {Component} from 'react-simplified';
import {Alert} from '../../src/widgets.js';
import {shallow, mount} from 'enzyme';
import sinon from 'sinon';
import {ChooseCategory} from "../../src/components/ChooseCategory/ChooseCategory";


describe('Test for ChooseCategory component', () => {
    const wrapper = shallow(<ChooseCategory />);


    it('renders correctly', () => {
        expect(wrapper).toMatchSnapshot();
    });

});