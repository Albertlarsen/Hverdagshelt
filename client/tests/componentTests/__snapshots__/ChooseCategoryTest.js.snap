// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test for ChooseCategory component renders correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ChooseCategory />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <ListGroup
          bsClass="list-group"
        />,
        <Col
          bsClass="col"
          componentClass="div"
          md={4}
        />,
        <Col
          bsClass="col"
          componentClass="div"
          md={4}
        >
          <p />
        </Col>,
        <Col
          bsClass="col"
          componentClass="div"
          md={4}
        />,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "bsClass": "list-group",
          "children": Array [],
        },
        "ref": null,
        "rendered": Array [],
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "bsClass": "col",
          "componentClass": "div",
          "md": 4,
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "bsClass": "col",
          "children": Array [
            <p />,
            undefined,
          ],
          "componentClass": "div",
          "md": 4,
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {},
            "ref": null,
            "rendered": null,
            "type": "p",
          },
          undefined,
        ],
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "bsClass": "col",
          "componentClass": "div",
          "md": 4,
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <ListGroup
            bsClass="list-group"
          />,
          <Col
            bsClass="col"
            componentClass="div"
            md={4}
          />,
          <Col
            bsClass="col"
            componentClass="div"
            md={4}
          >
            <p />
          </Col>,
          <Col
            bsClass="col"
            componentClass="div"
            md={4}
          />,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "bsClass": "list-group",
            "children": Array [],
          },
          "ref": null,
          "rendered": Array [],
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "bsClass": "col",
            "componentClass": "div",
            "md": 4,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "bsClass": "col",
            "children": Array [
              <p />,
              undefined,
            ],
            "componentClass": "div",
            "md": 4,
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {},
              "ref": null,
              "rendered": null,
              "type": "p",
            },
            undefined,
          ],
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "bsClass": "col",
            "componentClass": "div",
            "md": 4,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
